geode-web-api/src/main/java/org/apache/geode/rest/internal/web/util/JsonWriter.java

```diff
@@ -216,7 +216,8 @@ public static void writeArrayAsJson(JsonGenerator generator, Object value, Strin
       writeObjectArrayAsJson(generator, (Object[]) value, pdxField);
     } else {
       throw new IllegalStateException(
-          "PdxInstance returns unknwon pdxfield " + pdxField + " for type " + value);
+          "PdxInstance returns unsupported type " + value.getClass()
```

Comment:
pdxField can be null. Are you okay with this message just saying "for field null" in that case? I think this message should say that it is trying to write an array as JSON and found a "value" whose "type" can not be written as an array. The only caller I found of it already checked that value.getClass().isArray() is true so we just have an array that does not match all the checks this code does. I think you should test this with an array that will fail instead of a Date. Try a Date[]. 
By the way, the last "else if" on line 215 looks buggy to me. If you look at writeObjectArrayAsJson you will see it just iterates the array and delegates to writeValueAsJson which supports many more types than are check for here (for example an array of Link or an array of Map). If your array was declared as Object[], you could then put Link instances into it and this code would call writeObjectArrayAsJson. But if instead it declared it as Link[] I think this code would throw. It seems like it should work with any type of array as long as it contains values that writeValueAsJson can handle. But I'm not sure how the other direction works (going from JSON back to a java array).

Reply:
I am okay with just saying "for field null" if the field doesn't have a name. I've changed the message to:
`          "The pdx field " + pf + " is an array whose type " + value.getClass()
              + " can not be converted to JSON.");
`
Does that work?

As for the last "else if" on line 215, I agree it could be improved (similarly GEODE-9392 suggests there may be room to improve PdxToJSON.writeValue as well), but I'm trying to limit the scope here to making the existing messages more useful.

Reply:
I like your new messages